src/Uno.Extensions.Navigation.Toolkit/ServiceCollectionExtensions.cs

```diff
@@ -1,10 +1,39 @@
-﻿namespace Uno.Extensions;
+﻿using Uno.Extensions.Hosting;
+
+namespace Uno.Extensions;
 
 /// <summary>
 /// Extension methods for adding services to an <see cref="IServiceCollection" />.
 /// </summary>
 public static class ServiceCollectionExtensions
 {
+	private struct ToolkitViewInitializer : IRootViewInitializer
+	{
+		public ContentControl CreateDefaultView() => new ExtendedSplashScreen();
+
+		public void InitializeViewHost(FrameworkElement element, Task loadingTask)
+		{
+			if (element is LoadingView loadingView)
+			{
+				loadingView.Source = new LoadingTask(loadingTask, element);
+			}
+		}
+
+		public void PreInitialize(FrameworkElement element, IApplicationBuilder builder)
+		{
+			if (element is ExtendedSplashScreen splash)
+				splash.Initialize(builder.Window, builder.Arguments);
+		}
+	}
+
+	public static IApplicationBuilder AddToolkitNavigation(this IApplicationBuilder builder)
+	{
+		builder.Properties.Add(typeof(IRootViewInitializer), new ToolkitViewInitializer());
+		return builder.Configure(host => host.ConfigureServices(
+			 services => services.AddToolkitNavigation()));
```

Comment:
Don't call AddToolkitNavigation on the services, call UseToolkitNavigation on the hostbuilder